packages/plugins/typed-variable-modal/src/TypedVariableModal.js

```diff
@@ -0,0 +1,385 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Class responsible for creating a modal used for creating typed
+ *     variables.
+ * @author aschmiedt@google.com (Abby Schmiedt)
+ */
+
+
+import * as Blockly from 'blockly/core';
+import { Modal } from './Modal.js';
+
+/**
+ * Class for displaying a modal used for creating typed variables.
+ */
+export class TypedVariableModal extends Modal {
+  /**
+   * Constructor for creating a typed variable modal.
+   * @param {!Blockly.WorkspaceSvg} workspace The workspace that the modal will
+   *     be registered on.
+   * @param {string} btnCallbackName The name used to register the button
+   *     callback.
+   * @param {Array<Array<string>>} types An array holding arrays with the
+   *     display name as the first value and the type as the second.
+   *     Ex: [['Penguin', 'PENGUIN'], ['Giraffe', 'GIRAFFE']].
+   * @param {TypedVarModalMessages} opt_messages The messages for a typed variable
+   *     modal.
+   */
+  constructor(workspace, btnCallbackName, types, opt_messages) {
+    const title = opt_messages ? opt_messages["TYPED_VAR_MODAL_TITLE"] :
+        'Create Typed Variable';
+    super(title, workspace);
+
+    /**
+     * The id of the currently selected type.
+     * @type {?string}
+     * @private
+     */
+    this.selectedType_ = null;
+
+    /**
+     * The input div holding the name of the variable.
+     * @type {HTMLInputElement}
+     * @protected
+     */
+    this.variableNameInput_ = null;
+
+    /**
+     * The div holding the list of variable types.
+     * @type {HTMLElement}
+     * @protected
+     */
+    this.variableTypesDiv_ = null;
+
+    /**
+     * The first type input.
+     * @type {HTMLInputElement}
+     * @protected
+     */
+    this.firstTypeInput_ = null;
+
+    /**
+     * The name used to register the button callback.
+     * @type {string}
+     * @private
+     */
+    this.btnCallBackName_ = btnCallbackName;
+
+    /**
+     * An array holding arrays with the name of the type and the display name
+     *     for the type. Ex: [['Penguin', 'PENGUIN'], ['Giraffe', 'GIRAFFE']].
+     * @type {Array<Array<string>>}
+     * @protected
+     */
+    this.types_ = types;
+
+    const messages = opt_messages || {
```

Comment:
Might be better to do something like. 
Blockly.utils.object.mixin({
 "TYPED_VAR_MODAL_TITLE": "... ",
 ... 
}, opt_messages);

That way you ensure that we have at least the EN messages when querying Blockly.Msg['TYPED_VAR...']